### ISSUE TITLE
Fix clobbering of the upload size validation

### ISSUE THREAD

**Issue by nfauchelle** [silverstripe/silverstripe-framework#10058]
## Affected Version

4.8

## Description

yml config for the max size of uploads is getting over-written by a single '*' rule which is done in.
> src/Forms/UploadReceiver.php -> constructUploadReceiver() method.

We simply need to remove the comment and 3 lines that set the size from that method. 

If you review SilverStripe\Assets\Upload_Validator and check the getAllowedMaxFileSize method, you'll see the sizing will be populated (if it hasn't been done before).

yml config expected to work.

```yml
SilverStripe\Assets\Upload_Validator:
  default_max_file_size:
    '[image]': '2m'
    '*' : '1m'
```

## Steps to Reproduce

- Setup a new SilverStripe site.
- Add the config above to the yml file.
- Set your PHP to allow max post / max upload size of 10mb.
- Add the above config to your sites yml file and flush.
- In the CMS you'll be able to upload a 5MB file, when you shouldn't.

When you define max sizing in the yml file, it gets over-ridden with the CMS UploadField.

Here is the change here, https://github.com/nfauchelle/silverstripe-framework/commit/564ecbc6cbcfaee2e08c3655d182cf51c6a11605
I don't have permission to submit a pull request.

## PRs
- https://github.com/silverstripe/silverstripe-framework/pull/10059
- https://github.com/silverstripe/silverstripe-assets/pull/469
- https://github.com/silverstripe/silverstripe-asset-admin/pull/1430
- https://github.com/silverstripe/developer-docs/pull/430

**Comment by michalkleiner** [comment]
@nfauchelle thanks for reporting and the reproduction steps.

With PRs, generally, you'd fork the framework repo, fix the issue in a new branch off the latest minor version branch, i.e. `4.8` at this point in time, and then you can submit a cross-repo PR against that minor version branch from your fix branch.

**Comment by emteknetnz** [comment]
Linked PRs have been merged, they will be released in 5.2
